src/octoprint/templates/tabs/control.jinja2

```diff
@@ -1,10 +1,46 @@
+            <div class="btn btn-mini" data-bind="visible: document.pictureInPictureEnabled, click: function(data, event) { document.getElementById('webcam_webrtc').requestFullscreen() }"><i class="fas fa-expand"></i></div>
+            <div class="btn btn-mini" data-bind="visible: document.pictureInPictureEnabled, click: function(data, event) { document.getElementById('webcam_webrtc').requestPictureInPicture() }"><i class="fas fa-images"></i></div>
+            <div class="btn btn-mini" data-bind="click: function(data, event) { webcamMuted(!webcamMuted()); document.getElementById('webcam_webrtc').muted = webcamMuted() }">
+                <!-- ko if: webcamMuted() -->
+                <i class="fas fa-volume-mute"></i>
+                <!-- /ko -->
+                <!-- ko ifnot: webcamMuted() -->
+                <i class="fas fa-volume-up"></i>
+                <!-- /ko -->
+            </div>
+        </div>
+        <div data-bind="css: { webcam_rotated: settings.webcam_rotate90(), webcam_unrotated: !settings.webcam_rotate90() }">
+            <div class="rotation_target">
+                <div data-bind="css: { flipH: settings.webcam_flipH(), flipV: settings.webcam_flipV() }">
+                    <video id="webcam_webrtc" muted autoplay playsinline style="width: 100%"></video>
+                </div>
+            </div>
+        </div>
     </div>
     <div id="webcam_hls_container" tabindex="0" data-bind="visible: webcamHlsEnabled()">
-        <video id="webcam_hls" muted autoplay style="width: 100%"></video>
+        <div class="video-controls btn-group action-buttons">
+            <div class="btn btn-mini" data-bind="visible: document.pictureInPictureEnabled, click: function(data, event) { document.getElementById('webcam_hls').requestFullscreen() }"><i class="fas fa-expand"></i></div>
+            <div class="btn btn-mini" data-bind="visible: document.pictureInPictureEnabled, click: function(data, event) { document.getElementById('webcam_hls').requestPictureInPicture() }"><i class="fas fa-images"></i></div>
+            <div class="btn btn-mini" data-bind="click: function(data, event) { webcamMuted(!webcamMuted()); document.getElementById('webcam_hls').muted = webcamMuted() }">
+                <!-- ko if: webcamMuted() -->
+                <i class="fas fa-volume-mute"></i>
+                <!-- /ko -->
+                <!-- ko ifnot: webcamMuted() -->
+                <i class="fas fa-volume-up"></i>
+                <!-- /ko -->
+            </div>
+        </div>
+        <div data-bind="css: { webcam_rotated: settings.webcam_rotate90(), webcam_unrotated: !settings.webcam_rotate90() }">
+            <div class="rotation_target">
+                <div data-bind="css: { flipH: settings.webcam_flipH(), flipV: settings.webcam_flipV() }">
+                    <video id="webcam_hls" muted autoplay playsinline style="width: 100%"></video>
+                </div>
+            </div>
+        </div>
```

Comment:
All of this is pretty much duplicated. This screams for the use of a ko template to reduce duplication, and the logic in the click handlers should definitely go into descriptively named methods on the view model as well.

Reply:
I was actually wondering if we need that all twice...webrtc and HLS both use `<video>` so I think we can just have one set of elements for them. Are you aware of any reason we have this all twice?

Reply:
It was less duplication before, so it didn't feel as bad as now 😅 I'd suggest to either deduplicate it by reusing the same snippet for both types or at least use a template to not *actually* duplicate the source.

Reply:
Then let me not duplicate anything :D Sharing the video element caused some glitches when switching from HLS to webRTC, so I still have two of them but the rest is all shared.